compiler/damlc/daml-ide-core/src/Development/IDE/Core/Rules/Daml.hs

```diff
@@ -485,23 +486,30 @@ generatePackageMapRule opts =
         let hash = BS.concat $ map (T.encodeUtf8 . LF.unPackageId . LF.dalfPackageId) $ Map.elems res
         return (Just hash, ([], Just res))
 
-generateStablePackages :: FilePath -> IO ([FileDiagnostic], Map.Map (UnitId, LF.ModuleName) LF.DalfPackage)
-generateStablePackages fp = do
+generateStablePackages :: LF.Version -> FilePath -> IO ([FileDiagnostic], Map.Map (UnitId, LF.ModuleName) LF.DalfPackage)
+generateStablePackages lfVersion fp = do
     (diags, pkgs) <- fmap partitionEithers $ do
         allFiles <- listFilesRecursive fp
         let dalfs = filter ((== ".dalf") . takeExtension) allFiles
         forM dalfs $ \dalf -> do
-            let unitId = stringToUnitId $ takeFileName $ takeDirectory dalf
+            let packagePath = takeFileName $ takeDirectory dalf
+            let unitId = stringToUnitId $ if packagePath == "daml-stdlib"
+                    then damlStdlib -- We patch this to add the version number
+                    else packagePath
             let moduleName = LF.ModuleName (NonEmpty.toList $ T.splitOn "-" $ T.pack $ dropExtension $ takeFileName dalf)
             dalfPkgOrErr <- readDalfPackage dalf
             pure (fmap ((unitId, moduleName),) dalfPkgOrErr)
-    pure (diags, Map.fromList pkgs)
+    -- We filter out stable packages for newer LF versions, e.g., the stable packages for wrappers around Any.
+    -- It might seem tempting to make stable packages per LF version but this makes no sense at all.
+    -- Packages should remain stable as we move to newer LF versions.
```

Comment:
Maybe it's good to clarify that changing the LF version would change the hash. For me, that's the main reason why having stable packages per LF version doesn't make sense.

Reply:
Good idea, I’ve added a comment.